language/xsp/evaluation/official_evaluation.py

```diff
@@ -402,16 +399,23 @@ def execute_predictions(predictions, cache_dict, ofile, case_sensitive,
 
     # Get the gold results
     if cache_dict is None or gold_query not in cache_dict:
-      if verbose:
-        print('Trying to execute the gold query:\n\t' + gold_query)
-      gold_results, gold_exception_str = try_executing_query(
-          gold_query, cursor, case_sensitive, verbose)
-
-      if gold_exception_str:
-        gold_error += 1
-        gold_results = []
-      elif cache_dict is not None:
-        cache_dict[u''.join(gold_query).decode('utf-8')] = gold_results
+      if printable_utterance not in cache_dict:
+        print(gold_query)
+        print(printable_utterance)
+        raise ValueError('Cache miss!')
+
+        if verbose:
+          print('Trying to execute the gold query:\n\t' + gold_query)
+        gold_results, gold_exception_str = try_executing_query(
+            gold_query, cursor, case_sensitive, verbose)
+  
+        if gold_exception_str:
+          gold_error += 1
+          gold_results = []
+        elif cache_dict is not None:
+          cache_dict[u''.join(gold_query).decode('utf-8')] = gold_results
```

Comment:
The code after L405 is unreachable. Can we remove it or fix the code such that it can be reached when intended?

Reply:
Yep, I updated it so now it is an option to either update the cache or detect cache misses.